source/pele/base_potential_online.h

```diff
@@ -0,0 +1,88 @@
+#ifndef PELE_POTENTIAL_ONLINE_H
+#define PELE_POTENTIAL_ONLINE_H
+
+#include "pele/base_potential.h"
+
+namespace pele {
+    
+class BasePotentialOnline : public BasePotential {
+protected:
+    const size_t m_nr_terms;
+public:
+    BasePotentialOnline(const size_t nr_terms)
+        : m_nr_terms(nr_terms)
+    {}
+    virtual ~BasePotentialOnline() {}
+    virtual double get_energy(Array<double> x)
+    {
+        double energy = 0;
+        for (size_t i = 0; i < m_nr_terms; ++i) {
+            energy += get_energy(x, i);
+        }
+        return energy;
+    }
+    virtual double get_energy_ogradient(Array<double> x, const size_t index, Array<double> ograd)
```

Comment:
Maybe we can change this name?  Ideally names are self-documenting and "ogradient" does not really indicate what it does.

How about get_energy_gradient_online

or even better, get_energy_gradient_batch, and rename index to batch_number